aiopenapi3/v30/glue.py

```diff
@@ -77,33 +79,78 @@ def _prepare_security(self):
+            if flow := getattr(ss.flows, "implicit", None):
+                add_auths.append(httpx_auth.OAuth2Implicit(
+                    **value,
+                    authorization_url=flow.authorizationUrl,
+                    scopes=flow.scopes,
+                    # refresh_url=getattr(flow, "refreshUrl", None),
+                ))
+            if flow := getattr(ss.flows, "password", None):
+                add_auths.append(httpx_auth.OAuth2ResourceOwnerPasswordCredentials(
+                    **value,
+                    token_url=flow.tokenUrl,
+                    scopes=flow.scopes,
+                    # refresh_url=getattr(flow, "refreshUrl", None),
+                ))
+            if flow := getattr(ss.flows, "clientCredentials", None):
+                add_auths.append(httpx_auth.OAuth2ClientCredentials(
+                    **value,
+                    token_url=flow.tokenUrl,
+                    scopes=flow.scopes,
+                    # refresh_url=getattr(flow, "refreshUrl", None),
+                ))
+            if flow := getattr(ss.flows, "authorizationCode", None):
+                add_auths.append(httpx_auth.OAuth2AuthorizationCode(
+                    **value,
+                    authorization_url=flow.authorizationUrl,
+                    token_url=flow.tokenUrl,
+                    scopes=flow.scopes,
+                    # refresh_url=getattr(flow, "refreshUrl", None),
+                ))
+
+        if ss.type == "http":
+            if auth := auth_methods.get(ss.scheme_, None):
+                if isinstance(value, tuple):
+                    add_auths.append(auth(*value))
+                if isinstance(value, dict):
+                    add_auths.append(auth(**value))
+            if ss.scheme_ == "bearer":
+                add_auths.append(auth_methods["headerapikey"](
+                    f"{ss.bearerFormat or 'Bearer'} {value}", 
+                    "Authorization"
+                ))
+        
         value = cast(str, value)
-        if ss.type == "http" and ss.scheme_ == "bearer":
-            header = ss.bearerFormat or "Bearer {}"
-            self.req.headers["Authorization"] = header.format(value)
-
+        
         if ss.type == "mutualTLS":
             # TLS Client certificates (mutualTLS)
             self.req.cert = value
 
         if ss.type == "apiKey":
-            if ss.in_ == "query":
-                # apiKey in query parameter
-                self.req.params[ss.name] = value
-
-            if ss.in_ == "header":
-                # apiKey in query header data
-                self.req.headers[ss.name] = value
+            if auth := auth_methods.get((ss.in_+ss.type).lower(), None):
+                add_auths.append(auth(value, getattr(ss, "name", None)))
 
             if ss.in_ == "cookie":
                 self.req.cookies = {ss.name: value}
+        
+        for auth in add_auths:
+            if self.req.auth:
+                self.req.auth += auth
```

Comment:
this actually works?
I guess test_paths_security_combined should validate this

Reply:
Yes, although conditionally. It's a feature of httpx_auth. Ref: https://github.com/Colin-b/httpx_auth/blob/60bb9a7f4beb4b370398ae8fcb7d4402daa0b0bd/httpx_auth/authentication.py#L93

I think the code should be fine as is since all of that `auth` objects will be subclasses of `httpx_auth.authentication.SupportMultiAuth`, however I will add checks for this condition to be safe.